back/app/services/permissions/permissions_custom_fields_service.rb

```diff
@@ -36,32 +40,61 @@ def default_fields(permission)
     end
 
     # Add non-persisted locked fields to the permission if they don't exist & ensure they appear at the start of the list
-    def add_and_lock_related_fields(permission, fields, custom_field_ids, lock_type)
+    def add_and_lock_related_fields(permission, permission_custom_fields, custom_field_required_array, lock_type, insert_before: true)
```

Comment:
Maybe the `insert_before` parameter is redundant? It's always `false` when `lock_type` is `'group'`, and always `true` when `lock_type` is `'verification'`. So maybe simpler to just derive it from the `lock_type`?

Reply:
Good point. I did actually change it from that to make it less dependent on implementation, but actually looking at it now it seems over designed